src/js/modules/sections/selectors.js

```diff
@@ -72,3 +72,35 @@ export const getAllSectionLinksFromHome = createSelector(
     return sectionLinks;
   }
 );
+
+const getParentSections = (state) => Object.filter(getSections(state), sections => {
+    return sections.parentSlug == null;
+  });
+
+const getSubSections = (state) => Object.filter(getSections(state), sections => {
+    return sections.parentSlug !== null;
+  });
+
+
+export const makeSectionTree = createSelector(
+  [getParentSections, getSubSections],
+  (parentSectionObject, subSectionObject) => {
+    let sectionTree = [];
+    Object.keys(parentSectionObject).map((parent) => {
+      sectionTree.push({
+        name: parentSectionObject[parent].name,
+        slug: parentSectionObject[parent].slug,
+        subsections: Object.keys(subSectionObject).filter((subSection) => {
+          return subSectionObject[subSection].parentSlug == parentSectionObject[parent].slug
+        }).map((filteredSubSections) => {
```

Comment:
again, the key you're filtering by is not filteredSubSections. that name makes it sound like the key is an array or object. it should be subSectionSlug, I think